### PR TITLE
Pkcs11 prov

### PR DESCRIPTION
**Pull request by wbeck10**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: include/openssl/provider.h [hunk 1]
```diff
@@ -22,8 +22,9 @@ int OSSL_PROVIDER_set_default_search_path(OSSL_LIB_CTX *, const char *path);
 
 /* Load and unload a provider */
 OSSL_PROVIDER *OSSL_PROVIDER_load(OSSL_LIB_CTX *, const char *name);
-OSSL_PROVIDER *OSSL_PROVIDER_try_load(OSSL_LIB_CTX *, const char *name,
-                                      int retain_fallbacks);
+OSSL_PROVIDER *OSSL_PROVIDER_multiple_load(OSSL_LIB_CTX *, const char *provid, const char *name);
+OSSL_PROVIDER *OSSL_PROVIDER_try_load(OSSL_LIB_CTX *, const char *provid,
```
Comment (slontis) [inline include/openssl/provider.h:26]: You cannot modify a signature of an existing API.
To add additional params you need a new _ex method.

**Comment by slontis** [comment]
You will need to rebase..
@wbeck10 any chance you could coordinate with @simo5 to work towards a common goal?

**Comment by slontis** [comment]
FYI @beldmit

File: providers/pkcs11/pkcs11-v30/pkcs11.h [hunk 2]
```diff
@@ -0,0 +1,242 @@
+/* Copyright (c) OASIS Open 2016-2020. All Rights Reserved.
```
Comment (paulidale) [inline providers/pkcs11/pkcs11-v30/pkcs11.h:1]: This complicates the CLA situation I suspect.

**Comment by beldmit** [comment]
@wbeck10 I kindly ask you to look at this project: https://github.com/latchset/pkcs11-provider and probably participate in it

File: providers/pkcs11/pkcs11-v30/pkcs11.h (hunk 2 above)
Comment (simo5) [reply providers/pkcs11/pkcs11-v30/pkcs11.h:1]: Can you explain why?

Comment (t8m) [reply providers/pkcs11/pkcs11-v30/pkcs11.h:1]: We cannot add sources created by someone that we do not have CLA from. Regardless of the license of these sources.

Comment (simo5) [reply providers/pkcs11/pkcs11-v30/pkcs11.h:1]: Sounds like you should have a more nuanced stance wrt compatible licenses.
But not my problem.

**Comment by levitte** [comment]
If I may wish for something, it's a bit more story-telling in the commits.  Just titles makes it hard to review and to understand the reasoning behind some of the changes.

File: providers/pkcs11/pkcs11-v30/pkcs11.h (hunk 2 above)
Comment (wbeck10) [reply providers/pkcs11/pkcs11-v30/pkcs11.h:1]: Hi simo5, what would be the best way to contact you and talk about the PKCS11 provider project? Maybe we can work together?

Comment (simo5) [reply providers/pkcs11/pkcs11-v30/pkcs11.h:1]: We can discuss either on a pkcs11 provider issue, or you can email me, simo at redhat, if you prefer.

**Pull request closed without merging**